ramls/uuid.json

```diff
@@ -0,0 +1,5 @@
+{
+  "description": "A universally unique identifier (UUID), this is a 128-bit number used to identify a record and is shown in hex with dashes, for example 6312d172-f0cf-40f6-b27d-9fa8feaf332f; the UUID version must be from 1-5; see https://dev.folio.org/guides/uuids/",
+  "type": "string",
+  "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
```

Comment:
Please check if there are other usages of this regular expression and replace them with this schema

Reply:
I'll check request.json schema only. Or do you mean all the json schemes?

Reply:
I think it's worth checking all of them. This is effectively like extracting a method to remove duplication.

Reply:
Hi @marcjohnson-kint,

In the uuid.json I'm using pattern described here https://dev.folio.org/guides/uuids/.

But all existing UUIDs uses weaker pattern:
`^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$` - used now
`^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$` - used in `uuid.json`

I don't think that we can just replace all UUID patterns with the new one, it should be considered as a breaking change.

Should I maybe use the weak UUID pattern in the `uuid.json` file?

Reply:
> Hi @marcjohnson-kint,
> 
> In the uuid.json I'm using pattern described here https://dev.folio.org/guides/uuids/.

That is correct.

> But all existing UUIDs uses weaker pattern:
> `^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$` - used now
> `^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$` - used in `uuid.json`
> 
> I don't think that we can just replace all UUID patterns with the new one, it should be considered as a breaking change.

Agreed, this would be breaking compatibility. we shouldn't change any weak UUID regular expressions during this change.

> Should I maybe use the weak UUID pattern in the `uuid.json` file?

I don't think we should use the weak UUID pattern for the shared file.